spec/concerns/resource_url_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+class ResourceUrlController < ApplicationController
+  include ShiftCommerce::ResourceUrl
+
+  # override, as we don't have access to the real root_url here
+  def root_url
+    'http://localhost/'
```

Comment:
I don't know how to make `root_url` work properly in a controller test – I tried performing real requests, but the `root_url` helper still returned an `undefined method 'host'` error. As such, for simplicity's sake, I'm just overriding it here.